### DISCUSSION TITLE
Next.JS Maximum update depth exceeded

### DISCUSSION THREAD

**Discussion by edersonff** [pmndrs/zustand#2748] [Bug report]
### Bug Description

New version zustand^5.0.0-rc.2 is not working with Next.js, mainly when splitting apart the state.

Example:
`const [isLoading, setLoading] = useLoadingStore((state) => [
    state.loading,
    state.setLoading,
  ]);`

The error is:
> Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.

I downgraded to 4.5.5(last version used in a project) and worked just fine, maybe is a performance issue.

### Reproduction Link

https://stackblitz.com/edit/stackblitz-starters-bxkypz?file=app%2Fpage.tsx

**Comment by dai-shi** [comment]
`useShallow` should fix.

```diff
- const [isLoading, setLoading] = useLoadingStore((state) => [ state.loading, state.setLoading]);
+ const [isLoading, setLoading] = useLoadingStore(useShallow((state) => [ state.loading, state.setLoading]));
```

See also: #2741
